.github/workflows/bitcoin-tests.yml

```diff
@@ -18,8 +18,34 @@ concurrency:
   cancel-in-progress: ${{ github.event_name == 'pull_request' }}
 
 jobs:
+  generate-tests:
+    name: Generate JSON of tests to run
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+      ## Setup test environment
+      - name: Setup Test Environment
+        id: setup_tests
+        uses: stacks-network/actions/stacks-core/testenv@main
+        with:
+          btc-version: "25.0"
+      - name: Generate tests JSON
+        id: generate_tests_json
+        # List all of the tests using the nextest archive (so we don't need to do another build task)
+        # Filter them such that we only select tests from `--bin stacks-node` which are marked `ignored`
+        # Transform the output JSON into something that can be used as the matrix input
+        run: |
+          cargo nextest list --archive-file ~/test_archive.tar.zst -Tjson | \
+          jq -c '.["rust-suites"]["stacks-node::bin/stacks-node"]["testcases"] | map_values(select(.["ignored"] == true)) | keys' > ./tests.json
```

Comment:
I don't like that this prevents us from using `ignored` to actually ignore a test. What if we take this opportunity to use a different technique for these tests, and instead of marking them all as ignored, we only mark the ones that we want to ignore as ignored, and for the others we could use a new Cargo feature flag to ensure the simple CI job ignores them?

Reply:
The added benefit is that we would see directly at the source of the test that it really is being ignored.

Reply:
I'm not opposed to doing that, but I'd prefer to do it in a separate PR, because that would touch all the test definition's headers.

Reply:
Ok, so this PR aims to only change the yml file? I'm fine with that being a separate PR.

Reply:
Created #5802